chains/solana/contracts/programs/cctp-token-pool/src/lib.rs

```diff
@@ -693,6 +699,9 @@ impl CctpMessage {
 pub struct State {
     pub version: u8,
     pub config: BaseConfig,
+    // Authority allowed to reclaim funds (i.e. from closing the CCTP event PDA, or de-funding an
+    // overfunded signer PDA.)
+    pub fund_manager: Pubkey,
```

Comment:
We will need another method to extend the state PDA so that it fits this new field. Once migrated in staging, then we can remove the new method (no need to change the `version` field, this can be quick-and-dirty just for staging).
Also, if we are going to migrate, we could already add a allowed destinatary address for the withdrawed funds (similar to the router's `fee_aggregator` attribute), wdyt? And maybe also a `minimum_balance` field to ensure that the funds_manager never withdraws so much that the pools runs out of funds and leads to downtime. My thinking is maybe the owner sets the destinatary, and the funds manager is the one allowed to actually make the call to debit those funds while making (keeping the balance above a minimum).